src/lib/CopyBSR.svelte

```diff
@@ -0,0 +1,101 @@
+<script lang="ts">
+  import Fa from 'svelte-fa'
+  import { faTwitch } from '@fortawesome/free-brands-svg-icons'
+  import { onMount } from 'svelte'
+  import { fade } from 'svelte/transition'
+
+  export let fontSize: string = '1rem'
+  export let mapId: string
+
+  // Adding props to allow these to be customizable for the MOTW Section
+  export let tooltipTop: string = '32px'
+  export let tooltipLeft: string = '92%'
+  export let tooltipTransform: string = 'translateX(-50%)'
+
+  // Margins because it looks off on the MapCards without it
+  export let marginRight: string = '4.3px'
+  export let marginTop: string = '3px'
+
+  let showTooltip = false
+  let tooltipTimeout: NodeJS.Timeout
+
+  function copyBSR() {
+    const textToCopy = `!bsr ${mapId}`
+    navigator.clipboard
+      .writeText(textToCopy)
+      .then(() => {
+        showTooltip = true
+        clearTimeout(tooltipTimeout)
+        tooltipTimeout = setTimeout(() => {
+          showTooltip = false
+        }, 3000)
+      })
+      .catch((err) => {
+        console.error('Failed to copy BSR:', err)
+      })
+  }
+
+  onMount(() => {
+    return () => {
+      clearTimeout(tooltipTimeout)
+    }
+  })
```

Comment:
Two things here:

1. I'm not sure if you need to clear on mount. If components can remount with the previous state, then this is good. If no, it's just unnecessary. I'm not sure about Svelte details here.
2. I don't think this does anything as it is, unless Svelte uses function returns. But it seems to be it should rather be the following:

```ts
onMount(() => {
    clearTimeout(tooltipTimeout);
});
```

The inner function probably leads to it not being executed.

Reply:
I think you're right. I had some outside help on that part. Using your suggestion seems to work well.